include/deal.II/lac/la_parallel_block_vector.templates.h

```diff
@@ -937,10 +939,11 @@ namespace LinearAlgebra
       Assert (matrix.n() == n,
               dealii::ExcDimensionMismatch(matrix.n(),n));
 
-      (*this) = Number();
+      // we can't use sadd as we add multiple times to the same i-th block.
+      V *= s;
       for (unsigned int i = 0; i < n; i++)
         for (unsigned int j = 0; j < m; j++)
-          this->block(i).add (matrix(j,i), V.block(j));
+          V.block(i).add (matrix(j,i)*b, this->block(j));
```

Comment:
Instead of multiplication by `s`, I would peel off the first loop iterations of the `j` loop and use `sadd` there. I know it's one more line of code, but here I'd go for the faster variant.

Reply:
will fix